provider/image.go

```diff
@@ -146,6 +156,14 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		return "", nil, err
 	}
 
+	// add dockerfile to tarball if it's not in the build context
+	if dockerfileCtx != nil && tar != nil {
+		tar, build.Dockerfile, err = clibuild.AddDockerfileToBuildContext(dockerfileCtx, tar)
```

Comment:
Can we comment that this function injects the dockerfile and modifies .dockerignore, "see definition for details"?